src/sync.c

```diff
@@ -0,0 +1,79 @@
+#include "../include/my_blockchain.h"
+
+void read_nodes(int nid, char* bid, node_t **head){
+   
+    node_t *node = malloc(sizeof(node_t));
+    make_bid_buffer(bid, node);
+    node->nid = nid;
+    node->next = NULL;
+
+    if(*head == NULL){
+        *head = node;
+    }
+    else{
+        node_t *last_node = *head;
+
+        while(last_node->next != NULL)
+            last_node = last_node->next;
+        
+        last_node->next = node;
+    }
+}
+
+node_t *get_synced_nodes(node_t *head){
+    
+    int file = open("./bin/saved_nodes.txt", O_RDWR),
+        skip = 0;
+
+    char    *temp = malloc(sizeof(char)),
+            *nid = malloc(sizeof(char)*100),
+            *bid = malloc(sizeof(char)*100);
```

Comment:
Do we need to allocate these in the heap?  Are these buffers local to this function? If, the function `make_bid_buffer(bid, node)` is already allocated for the `bid` buffer as you create a new node in the list.

Reply:
This is only for the function so we save the data there and then add it to the list